toolchain/semantics/semantics_ir_formatter.cpp

```diff
@@ -0,0 +1,584 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/semantics/semantics_ir_formatter.h"
+
+#include "llvm/ADT/StringExtras.h"
+#include "llvm/ADT/StringSet.h"
+#include "llvm/Support/SaveAndRestore.h"
+#include "toolchain/lexer/tokenized_buffer.h"
+#include "toolchain/parser/parse_tree.h"
+
+namespace Carbon {
+
+namespace {
+// Assigns names to nodes, blocks, and scopes in the Semantics IR.
```

Comment:
Once this stabilizes a bit, expand this with documentation of the naming scheme and some of the rationale behind it?

Alternatively, if creating a standalone documentation file for the textual format, could document it there and just have a reference here to those docs.

Reply:
I'll add documentation for the textual IR format as a follow-up.